pkg/dataobj/consumer/partition_processor.go

```diff
@@ -339,18 +338,22 @@ func (p *partitionProcessor) processRecord(record *kgo.Record) {
 	p.lastModified = p.clock.Now()
 }
 
-func (p *partitionProcessor) commitRecords(record *kgo.Record) error {
+// commits the offset of the last record processed. It should be called after
+// each successful flush to avoid duplicate data in consecutive data objects.
+func (p *partitionProcessor) commit() error {
+	if p.lastRecord == nil {
```

Comment:
This should never happen because of other checks, but it is a bit of defensive programming.